marko.json

```diff
@@ -37,12 +37,17 @@
         "@index": "string",
         "@type": "string",
         "@slide": "string",
+        "@autoplay": "boolean",
+        "@paused": "boolean",
```

Comment:
Setting a `paused` property feels a little odd to me. I would have thought we'd use `play()` and `pause()` methods similar to the HTML video tag. I would imagine a `paused` attribute would be read-only (again, like the video tag) if we are trying to be HTML-like. @senthilp thoughts?

Reply:
I took inspirited from the video tags `muted` property. We could expose methods for this as well if we want though.

Reply:
Also I am exposing the stateful `paused` property on the element (which the video tag also does). However the video element exposes 10^10 ways of configuring it's state so 🤷‍♂️. Previously I think we decided against having methods for boolean toggles like this.

Reply:
Stateful `paused` makes sense, very similar to `video` tag's `paused`. This is sort of an in-between situation where there is no right or wrong.

Reply:
@senthilp Video tag's `paused` is read-only.

Reply:
... and of course this is a carousel, not video, I know I know ;-) But trying to find some common ground.

Reply:
I think `muted` is just an attribute for initial render, and not stateful:

`muted - Whether to mute the media resource by default`

https://www.w3.org/TR/html51/semantics-embedded-content.html#the-video-element

Reply:
I am okay with either option. Making `paused` not stateful and switching to `play()` and `pause()` methods is also fine, could make the docs slightly more confusing though.

Reply:
But `paused` will make no sense on 99% of our carousels, right? Only the ones with autoplay.

Reply:
That is correct.

Reply:
I would remove `paused` in order to fix this anomaly.

Reply:
Actually, let me think on that last comment a bit more ;-)

Reply:
My conclusion is... that they should have named it `automute` rather than `muted`. Lol.

I can see the attraction and simplicity of just a single stateful property to control playback (especially as we have the power to do that in our Marko layer on top of HTML), but if we are striving to be like the HTML API then we should keep `paused` as a readonly property, and add play() and pause() methods. This will be most familiar for users of the video tag. Not saying it's better, just more familiar ;-)

In all honesty I don't mind either way, but it feels like as we've been strict with the API up to now, we may as well continue on this path. I think after v1 we should asses all of the components again (properties vs methods) as a whole and see if it's still the right approach.

Reply:
Are we fine with allowing `paused` to be set programmatically as attributes when using marko but just removing the property? Otherwise we are stuck with changing the state imperatively which is lame 😕.

Reply:
@ianmcburnie I tried the `muted` attribute on YouTube `video` tag and it is stateful. I was able to mute and unmute by doing `$0.muted = true` and `$0.muted = false` 😄 .

Reply:
Hmm, that didn't work for me in Safari. Let me try again...

Reply:
Adding and removing the attribute is not stateful though. Is that what you meant? Only setting it programmatically works for me (i.e. $0.muted = true and $0.muted = false).

Reply:
Yes, adding/removing does not work. Only setting it programmatically works.

Reply:
I think `paused` is fine as a property since it has a direct (and useful) mapping to the state. Previously we discussed using methods for either 1) dealing with non-primitive data types, or 2) acting on the component without a clear mapping to state.